Sources/MessagesViewController.swift

```diff
@@ -240,15 +243,23 @@ extension MessagesViewController: UICollectionViewDataSource {
 
 extension MessagesViewController {
 
+
     fileprivate func addKeyboardObservers() {
         NotificationCenter.default.addObserver(self, selector: #selector(handleKeyboardDidChangeState), name: .UIKeyboardWillChangeFrame, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(handleTextViewDidBeginEditing), name: .UITextViewTextDidBeginEditing, object: messageInputBar.inputTextView)
```

Comment:
Out of interest, why are you choosing to add an observer here over assigning the delegate of the textView?

Reply:
I thought it would be better to not reserve making the delegate our controller